Source/Core/Core/AchievementManager.cpp

```diff
@@ -423,10 +423,25 @@ AchievementManager::GetUnlockStatus(AchievementId achievement_id) const
   return m_unlock_map.at(achievement_id);
 }
 
-void AchievementManager::GetAchievementProgress(AchievementId achievement_id, u32* value,
-                                                u32* target)
+AchievementManager::ResponseType
+AchievementManager::GetAchievementProgress(AchievementId achievement_id, u32* value, u32* target)
 {
-  rc_runtime_get_achievement_measured(&m_runtime, achievement_id, value, target);
+  if (!IsGameLoaded())
+  {
+    ERROR_LOG_FMT(
+        ACHIEVEMENTS,
+        "Attempted to request measured data for achievement ID {} when no game is running.",
+        achievement_id);
+    return ResponseType::INVALID_REQUEST;
+  }
+  int result = rc_runtime_get_achievement_measured(&m_runtime, achievement_id, value, target);
+  if (result != 1)
```

Comment:
The [documentation](https://github.com/RetroAchievements/rcheevos/wiki/rc_runtime_get_achievement_measured) discriminates the return values as 'zero' and 'non-zero', so you should check for zero here.